dawn/renderer/components/Editor.js

```diff
@@ -300,136 +284,107 @@ class Editor extends React.Component {
               onClick={this.startRobot}
               glyph="play"
               disabled={this.props.isRunningCode || !this.props.runtimeStatus}
             />
-            <EditorButton
+            <TooltipButton
               id="stop"
               text="Stop"
               onClick={this.stopRobot}
               glyph="stop"
               disabled={!(this.props.isRunningCode && this.props.runtimeStatus)}
             />
-            <EditorButton
+            <TooltipButton
               id="upload"
               text="Upload"
               onClick={this.upload}
               glyph="upload"
               // disabled={this.props.isRunningCode || !this.props.runtimeStatus}
             />
+            <TooltipButton
+              id="e-stop"
+              text="E-STOP"
+              onClick={this.estop}
+              glyph="fire"
+            />
           </ButtonGroup>
           <ButtonGroup id="console-buttons">
-            <EditorButton
+            <TooltipButton
               id="toggle-console"
               text="Toggle Console"
               onClick={this.toggleConsole}
               glyph="console"
             />
-            <EditorButton
+            <TooltipButton
               id="clear-console"
               text="Clear Console"
               onClick={this.props.onClearConsole}
               glyph="remove"
             />
           </ButtonGroup>
-          <ButtonGroup id="misc-buttons">
-            <EditorButton
-              id="e-stop"
-              text="E-STOP"
-              onClick={this.estop}
-              glyph="fire"
-            />
-            <EditorButton
+          <ButtonGroup id="editor-settings-buttons">
+            <TooltipButton
               id="increase-font-size"
               text="Increase font size"
               onClick={this.increaseFontsize}
               glyph="zoom-in"
               disabled={this.props.fontSize > 28}
             />
-            <EditorButton
+            <TooltipButton
               id="decrease-font-size"
               text="Decrease font size"
               onClick={this.decreaseFontsize}
               glyph="zoom-out"
               disabled={this.props.fontSize < 7}
             />
-            <EditorButton
-              id="updates"
-              text="Updates"
-              onClick={this.toggleUpdateModal}
-              glyph="cloud-upload"
-            />
-            <EditorButton
-              id="configuration"
-              text="Configuration"
-              onClick={this.toggleConfigModal}
-              glyph="cog"
-            />
+            <DropdownButton
+              title="Theme"
+              bsSize="small"
+              id="choose-theme"
+            >
+              {_.map(this.themes, (theme, index) => (
+                <MenuItem
+                  active={theme === this.props.editorTheme}
+                  onClick={_.partial(this.changeTheme, theme)}
+                  key={index}
```

Comment:
Don't use `index` as a key; use the theme name or something unique/theme instead. (I believe React also complains in the console about this issue.)